docs/docs/basic-usage.md

```diff
@@ -195,7 +195,8 @@ the dependencies installed are still working even if your dependencies released
 
 ### Installing dependencies only
 
-The current project is installed in [editable](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs) mode by default.
+The current package is installed in [editable mode](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs) by default if it matches the canonical form of the project's name in `pyproject.toml`.
```

Comment:
I think these are slightly different concerns. The project is still installed in editable mode.

However, by virtue of the project not having the same name as the packages it contains and/or project follows a `src` layout, poetry does not auto-detect the packages and hence it does not get added to the `.pth` file. This is why the project is not importable, especially when the package is not at the project root (since this means it is not in PYTHON_PATH either, which automatically contains current working directory - `python -m site`).

I am not sure if this is the right place to clarify this. That said, off the top of my head, I am not sure where best to put it either. :disappointed: I feel this is a common enough issue that it needs to be addressed somewhere in the docs.